Sources/AsyncHTTPClient/HTTPHandler.swift

```diff
@@ -829,9 +843,11 @@ extension TaskHandler: ChannelDuplexHandler {
                 // All writes have to be switched to the channel EL if channel and task ELs differ
                 if context.eventLoop.inEventLoop {
                     context.writeAndFlush(self.wrapOutboundOut(.body(part)), promise: promise)
+                    self.actualBodyLength += part.readableBytes
                 } else {
                     context.eventLoop.execute {
                         context.writeAndFlush(self.wrapOutboundOut(.body(part)), promise: promise)
+                        self.actualBodyLength += part.readableBytes
```

Comment:
these two lines need to be flipped. `writeAndFlush` runs essentially arbitrary code (from the pipeline but also from the `promise`. So you need to do the state modification before that, or else you can get reordering.

Same a few lines above.

Reply:
fixed, thanks!